pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -174,18 +176,22 @@ func (r *ReconcileControlPlane) Reconcile(request reconcile.Request) (reconcile.
 			return reconcile.Result{}, nil
 		}
 		reqLogger.Info("Deleting ServiceMeshControlPlane")
-		result, err := reconciler.Delete()
+		err := reconciler.Delete()
 		// XXX: for now, nuke the resources, regardless of errors
```

Comment:
Shouldn't we return early and requeue if deleting fails? It seems wrong to me to try to remove the finalizer if there are still resources lurking about. Or do we explicitly rely on the GC in case of error?

Reply:
(note that this was not changed in this PR so this comment is just a general question)

Reply:
Yup, I was thinking the same thing. I'll have a PR shortly.